### ISSUE TITLE
Rule: remote-write.config should not be the trigger to enable stateless ruler

### ISSUE THREAD

**Issue by ahurtaud** [thanos-io/thanos#6469]
### Is your proposal related to a problem?

The flag `--remote-write.config` _automatically enables stateless mode for ruler and no series will be stored in the ruler's TSDB._ 

**Problem**: This makes impossible for a stateful ruler to remote_write a subset of their generated recording rule metrics to an external system.
This is something we are currently doing in prometheus and impossible to achieve in stateful ruler.
  - Create some recording rules, remote_write only matching series to an external remote-write capable system.

### Describe the solution you'd like

I would like to have a dedicated flag to enable stateless mode (and also enforce a remote_write config when it is set to true), so that I can still execute recording rules in stateful mode and remote_write some series to an external system.

### Describe alternatives you've considered

Migrating the rulers to stateless?: We are already using this architecture for some other rulers but I would like to not use stateless mode for this usecase.

**Comment by yeya24** [comment]
Sounds good! I agree a dedicated flag is better. Help wanted

**Comment by manulpatel** [comment]
Hello @yeya24! I am new to Thanos and would like to work on this issue and make myself familiar with the codebase. Could you please give some hints to get started to develop this feature?

**Comment by moki1202** [comment]
@ahurtaud @yeya24 I think I can work on this issue. This would be my approach here. Define stateless mode flag in `rule.go` and handle the effects in `RegisterPathOrContent` function so that when the stateless flag is set to false, both the file path and content flags will be ignored (I can't figure out how stateful mode is handled in terms of time series data storage) and similarly when its set to true everything works as usual. Please let me know if this approach works.

**Comment by yeya24** [comment]
> Define stateless mode flag in rule.go and handle the effects in RegisterPathOrContent function so that when the stateless flag is set to false, both the file path and content flags will be ignored (I can't figure out how stateful mode is handled in terms of time series data storage)

I don't think we need to change anything in `RegisterPathOrContent` function. Just need an `if else` statement to  use the path only if the flag is enabled.

> so that I can still execute recording rules in stateful mode and remote_write some series to an external system

This might be tricky to implement. Now we can either enable remote write or tsdb but this feature request requires to enable them both. I don't think it is as easy as adding a new flag.

A better way to describe the feature request to me is to support remote write in stateful mode to send metrics somewhere. @ahurtaud Does it sound correct?

**Comment by moki1202** [comment]
Understood! I'll open a PR soon.

**Comment by aditya7302** [comment]
@yeya24 If this issue is still open I would like to work on it. Can you assign it to me?
